### ISSUE TITLE
Travis-CI failure with .NET Core

### ISSUE THREAD

**Issue by nightblade9** [Thraka/SadConsole#142]
Hi,

You may remember me as [the person who asked for .NET Core support](https://github.com/Thraka/SadConsole/issues/137). I finally got around to unit-testing my game and decided to add Travis-CI integration.

TL;DR it failed with something related to MonoGame. Can you please tell me if/what I'm doing wrong? It looks like it's failing to build the project.

The core error appears to be:

```
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018: The "GetMGCBTask" task failed unexpectedly. 
```

The full stack:

```
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018: The "GetMGCBTask" task failed unexpectedly. [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018: System.IO.InvalidDataException: Central Directory corrupt. ---> System.IO.IOException: Invalid argument [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at System.IO.FileStream.CheckFileCall(Int64 result, Boolean ignoreNotSupported) [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at System.IO.FileStream.Seek(Int64 offset, SeekOrigin origin) [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at System.IO.Compression.ZipArchive.ReadEndOfCentralDirectory() [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    --- End of inner exception stack trace --- [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at System.IO.Compression.ZipArchive.ReadEndOfCentralDirectory() [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at System.IO.Compression.ZipArchive.Init(Stream stream, ZipArchiveMode mode, Boolean leaveOpen) [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at System.IO.Compression.ZipArchive..ctor(Stream stream, ZipArchiveMode mode, Boolean leaveOpen, Encoding entryNameEncoding) [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at System.IO.Compression.ZipFile.Open(String archiveFileName, ZipArchiveMode mode, Encoding entryNameEncoding) [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at System.IO.Compression.ZipFile.ExtractToDirectory(String sourceArchiveFileName, String destinationDirectoryName, Encoding entryNameEncoding, Boolean overwrite) [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at MSBuildTasks.GetMGCBTask.Execute() in /home/harry/GitHub/MonoGame.Content.Builder/Tasks/GetMGCBTask.cs:line 38 [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute() [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(27,5): error MSB4018:    at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask) [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(92,5): error MSB3073: The command "mono "/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MGCB/build/MGCB.exe" /@:"/home/travis/build/nightblade9/escape-to-earth/EscapeToEarth/Content/Content.mgcb" /platform:DesktopGL /quiet /outputDir:"bin/DesktopGL/Content" /intermediateDir:"obj/DesktopGL/Content"" exited with code 127. [/home/travis/build/nightblade9/escape-to-earth/EscapeToEarth/EscapeToEarth.csproj]
```

Travis log [here](https://api.travis-ci.org/v3/job/416451867/log.txt).

Travis configuration:

```
language: csharp
solution: escape-to-earth.sln
mono: none
dotnet: 2.2.200
script:
  cd EscapeToEarth.UnitTests
dotnet test
```

Project repository lives [here](https://github.com/nightblade9/escape-to-earth). I can build this locally and run tests on Windows 8/10.

**Comment by Thraka** [comment]
Have you installed the MonoGame SDK?

**Comment by nightblade9** [comment]
Yes. I don't know where it came from (probably the starter `SadConsole` project I copied which you zipped). Here are the contents today:

```

#----------------------------- Global Properties ----------------------------#

/outputDir:bin/$(Platform)
/intermediateDir:obj/$(Platform)
/platform:DesktopGL
/config:
/profile:Reach
/compress:False

#-------------------------------- References --------------------------------#


#---------------------------------- Content ---------------------------------#


```

**Comment by Thraka** [comment]
Yeah me neither. It does seem like some problem with that monogame tool. If anything you could remove the reference from the build targets so it doesn't run (if you don't need it to run, that tool converts assets to XNB format, which SadConsole doesn't use)

**Comment by Thraka** [comment]
According to this: https://stackoverflow.com/questions/20960403/c-sharp-end-of-central-directory-record-could-not-be-found it seems the error means that it is trying to open the file as a zip file, but it's not.

**Comment by nightblade9** [comment]
I removed the reference to the `Content.mgcb` file from the project and rebuilt it. I think I'm getting a similar/same error now for SadConsole (which I'm building from source against .NET Core 2.1):

```
/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MonoGame.Content.Builder.targets(92,5): error MSB3073: The command "mono "/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MGCB/build/MGCB.exe" /@:"/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/Content/Content.mgcb" /platform:DesktopGL /quiet /outputDir:"bin/DesktopGL/Content" /intermediateDir:"obj/DesktopGL/Content"" exited with code 127. [/home/travis/build/nightblade9/escape-to-earth/SadConsoleDotNetCore20/SadConsole.csproj]
```

([Source](https://travis-ci.org/nightblade9/escape-to-earth/builds/416945223?utm_source=email&utm_medium=notification))

I assume I can't just apply the same change, because SadConsole probably depends on something in the content pipeline for fonts.

What should I do?

**Comment by Thraka** [comment]
Nope, SadConsole just loads direct PNG files and doesn't use the pipeline. I just looked up linux exit code 127 and it says that it really means file not found. https://stackoverflow.com/questions/1763156/127-return-code-from

Does `/home/travis/.nuget/packages/monogame.content.builder/3.7.0.4/build/MGCB/build/MGCB.exe` exist?

**Comment by nightblade9** [comment]
I don't know. Travis workspaces are still a mystery to me (I don't know how to view/save files on disk).

Instead, I did some googling on how to get MonoGame projects to build on Travis. I found [this wonderful answer](https://stackoverflow.com/questions/37048232/how-to-build-a-monogame-c-sharp-project-on-travis-ci); the summary is that I added these changes to my `.travis.yml` file:

```
before_install:
  - wget http://www.monogame.net/releases/v3.4/MonoGame.Linux.zip
  - unzip MonoGame.Linux.zip
  - find * -type f -exec chmod 777 {} \; # lazy just make all executable
install:
  - sudo apt-get install -qq monodevelop libopenal-dev ttf-mscorefonts-installer gtk-sharp2; # install needed packages for the monogame sdk
  - sudo ./generate.sh && yes Y|sudo ./monogame-linux.run; # install monogame sdk
```

(This is a slightly modified version because one of the packages, erm, I think it doesn't exist in the version of Linux that the Travis docker container is using.)

This resulted in a successful build, so I'm good to go now.  Thanks for your help.
